### ISSUE TITLE
Pop-up (modal) windows are above fullscreened windows

### ISSUE THREAD

**Issue by vinhor** [paperwm/PaperWM#902]
**Describe the bug**
Dialog windows (e. g. Déjà Dup's backup process windows) overflow to other windows and don't hide even when you watch videos in full-screen
**To Reproduce**
1. Open app with dialog windows (e. g. Déjà DUp, YaST software)
2. Make the dialog window open
3. Try to watch video in fullscreen
4. The window won't hide

**Expected behavior**
The pop-up window will stay on top of the mother app and will hide when watching something in full-screen
**Screenshots**
![Screenshot from 2024-07-20 12-21-48](https://github.com/user-attachments/assets/326942f5-ba1f-4adb-b29b-f923a47a7a7a)
**System information:**
Distribution: openSUSE Tumbleweed
GNOME Shell: 46.3.1
Display server: Wayland
PaperWM version: 46.13.5
Enabled extensions:
- AlphabeticalAppGrid@stuarthayhurst
- app-hider@lynith.dev
- blur-my-shell@aunetx
- caffeine@patapon.info
- color-picker@tuberry
- custom-hot-corners-extended@G-dH.github.com
- dash-to-dock@micxgx.gmail.com
- unredirect@vaina.lt
- emoji-copy@felipeftn
- HideItems@fablevi.github.io
- hidetopbar@mathieu.bidon.ca
- paperwm@paperwm.github.com
- drive-menu@gnome-shell-extensions.gcampax.github.com
- user-theme@gnome-shell-extensions.gcampax.github.com
- Vitals@CoreCoding.com
- clipboard-history@alexsaveau.dev

**Comment by Lythenas** [comment]
I think this is unfortunately just a downside of the hacks we have to do to get tiling to work.

You could try enabling "Attach Model Dialogs" in Gnome Tweaks. I would expect that this breaks some things but it might work for some cases.

**Comment by jtaala** [comment]
> I think this is unfortunately just a downside of the hacks we have to do to get tiling to work.
> 
> You could try enabling "Attach Model Dialogs" in Gnome Tweaks. I would expect that this breaks some things but it might work for some cases.

That's correct, see README.md for settings that are managed by PaperWM due to incompatibility with the tiling model used in PaperWM:

https://github.com/paperwm/PaperWM?tab=readme-ov-file#recommended-gnome-shell-settings

Note, that enabling "Attach Modal Dialogs" won't work in PaperWM, as PaperWM will disable that setting when it's being enabled (and restore that setting to it's original state when PaperWM is disabled).

**Comment by jtaala** [comment]
P.S. we should probably change the README.md section header for these settings from _Recommended Gnome Shell Settings_ to _Managed Gnome Shell Settings_.

**Comment by jtaala** [comment]
@vinhor,

Not sure there's much we can do re `attach modal dialogs` support - but if the issue is hiding modal dialogs in fullscreen, we could have PaperWM explicity address modal dialogs when a window goes fullscreen (e.g. hide the dialog or disable it's `always on top` parameter etc.).

**Comment by jtaala** [comment]
Actually, setting the "always on top" parameter for fullscreen windows looks like it will put it self above any other modal dialogs...

We could have PaperWM set a window to "always on top" when fullscreening it.

**Comment by jtaala** [comment]
All,

I've implemented a small change in PR #903 that ensures fullscreened windows will be "above" modal dialogs.  Feel free to check it out and give it a try:

```
git fetch --all
git checkout improve-fullscreen-windows-set-always-on-top
./install.sh
```

then logout / login.

**Comment by jtaala** [comment]
P.S. I'll update this issue title re fullscreen windows above modal windows.

We can always create a feature request to re-look at supporting `Attach modal dialogs` (let me know is anyone if keen to give that one a go and create a PR for it!).
